From: Mattia Dongili <malattia@linux.it>
Date: Sun, 19 Feb 2006 12:50:37 +0000 (+0100)
Subject: kbuild: fix a cscope bug (make cscope segfaults)
X-Git-Url: http://www.kernel.org/git/?p=linux/kernel/git/sam/kbuild.git;a=commitdiff;h=0f558c333445a0181f33f9e6f996ce7cf008206a

https://bugzilla.novell.com/show_bug.cgi?id=152643



kbuild: fix a cscope bug (make cscope segfaults)

Workaround a cscope bug where a trailing ':' in VPATH makes it segfault
and let it build the cross-reference succesfully.

VPATH=/home/mattia/devel/kernel/git/linux-2.6: cscope -b
[1]    17555 segmentation fault VPATH=/home/mattia/devel/kernel/git/linux-2.6: cscope -b

Signed-off-by: Mattia Dongili <malattia@linux.it>
Signed-off-by: Sam Ravnborg <sam@ravnborg.org>
Signed-off-by: Olaf Hering <olh@suse.de>
---

--- a/Makefile
+++ b/Makefile
@@ -137,7 +137,7 @@ objtree		:= $(CURDIR)
 src		:= $(srctree)
 obj		:= $(objtree)
 
-VPATH		:= $(srctree):$(KBUILD_EXTMOD)
+VPATH		:= $(srctree)$(if $(KBUILD_EXTMOD),:$(KBUILD_EXTMOD))
 
 export srctree objtree VPATH TOPDIR
 
